src/pages/index.js

```diff
@@ -1,67 +1,128 @@
+  });
+};
+
+enableValidation(validationSettings);
+
+async function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+  try {
+    popupInstance.renderLoading(true, loadingText);
+    await request();
+    popupInstance.close();
+  } catch (err) {
+    console.error("Error in handleSubmit:", err);
+  } finally {
+    popupInstance.renderLoading(false);
+  }
+}
 
 // USER INFO INIT
 const userInfo = new UserInfo({
   nameSelector: profileName,
-  jobSelector: profileDescription,
+  aboutSelector: profileAbout,
+  avatarSelector: profileAvatar,
 });
 
-function handleEditFormSubmit(data) {
-  userInfo.setUserInfo({
-    name: data.title,
-    job: data.description,
-  });
+let currentUserId = null;
+
+function handleEditFormSubmit(inputValues) {
+  async function makeRequest() {
+    console.log("Updating profile with values:", inputValues);
+    const userData = await api.updateProfile({
+      name: inputValues.name,
+      about: inputValues.about,
+    });
+    userInfo.setUserInfo(userData);
+  }
+  return handleSubmit(makeRequest, profileEditPopup);
+}
 
-  profileEditForm.reset();
-  profileEditFormValidator.disableButton();
-  profileEditPopup.close();
+function handleAddFormSubmit(inputValues) {
+  async function makeRequest() {
+    console.log("Adding card with values:", inputValues);
+
+    const cardData = await api.addCard(inputValues);
+    const cardElement = createCard(cardData, currentUserId);
+    cardSection.addItem(cardElement);
+  }
+  return handleSubmit(makeRequest, profileAddPopup);
 }
 
-function handleAddFormSubmit(data) {
-  const cardData = {
-    place: data.place,
-    link: data.link,
-    alt: `Image of ${data.place}`,
-  };
-  const cardElement = createCard(cardData);
-
-  cardSection.addItem(cardElement);
-  profileAddForm.reset();
-  profileAddFormValidator.disableButton();
-  profileAddPopup.close();
+function handleAvatarFormSubmit(inputValues) {
+  async function makeRequest() {
+    console.log("Updating avatar with values:", inputValues);
+    const userData = await api.updateAvatar(inputValues.link);
+    userInfo.setUserInfo(userData);
+  }
+  return handleSubmit(makeRequest, profileAvatarPopup);
+}
+
+function handleCardDelete(card) {
+  confirmationPopup.open(() => {
+    card
+      .deleteCard()
+      .then(() => {
+        confirmationPopup.close();
+      })
+      .catch((err) => {
+        console.error("Error deleting card:", err);
+      });
+  });
+}
+
+async function handleLikeButton(card) {
+  try {
+    let updatedCard;
+    if (card._isLiked) {
+      // UNLIKE
+      updatedCard = await api.unlikeCard(card._id);
+    } else {
+      // LIKE
+      updatedCard = await api.likeCard(card._id);
+    }
+
+    if (updatedCard.isLiked !== undefined) {
+      card._isLiked = updatedCard.isLiked;
+      if (card._isLiked) {
+        card._likeButton.classList.add("card__like-button_active");
+      } else {
+        card._likeButton.classList.remove("card__like-button_active");
```

Comment:
Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle